### ISSUE TITLE
Clarification on DataInf for Influential data identification

### ISSUE THREAD

**Issue by ajsanjoaquin** [ykwon0407/DataInf#2]
Hello, thanks again for uploading the notebooks I requested! I am trying to adapt your method to some of my experiments.

I have a question about your proposed method: for calculating the influence of a training point `z` on the prediction of a validation point `z'`, why use the [average of the validation gradients](https://github.com/ykwon0407/DataInf/blob/82eadf780c5b57ac6c09a6469ac08a52ab8093eb/src/influence.py#L46) instead of the gradient of `z'` only?

**Comment by ykwon0407** [comment]
Thank you for the question. To clarify, The influence function can be applied to both the average validation loss and a validation loss evaluated at a particular point. We considered the average loss for the mislabeled data detection task and an individual validation loss for the influential data identification task. The influence function can be defined for both tasks, but they yield different meanings.
- When you use the average validation loss, your influence function measures "the impact of a training data point on the average validation loss" 
- However, when you use a particular validation point, say `z'`, it means "the impact of a training data point on a validation loss evaluated at `z'`". This is actually a special case of the average validation loss when there is only one data point. 

Then, you may ask "Which one should I look at? Average validation loss or one validation loss?" It depends on your scientific questions. If you want to see the impact on the test accuracy, then the average validation loss can be a good proxy. If you want to understand a model prediction, a validation loss at a point `z'` can be more effective.

**Comment by ajsanjoaquin** [comment]
Sorry, I misread your notebook. I didn't notice that you used  you used `IFEngineGeneration`   instead of `IFEngine` class in the Influential data identification task, which does not calculate the influence values on based on the average validation gradients.

**Comment by ykwon0407** [comment]
No problem! Thanks :)
